tests/test_filters.py

```diff
@@ -741,6 +741,42 @@ class OrderingListView(generics.ListAPIView):
 
         reload_module(filters)
 
+    def test_ordering_with_overridden_get_serializer_class(self):
+        class OrderingListView(generics.ListAPIView):
+            queryset = OrderingFilterModel.objects.all()
+            filter_backends = (filters.OrderingFilter,)
+            ordering = ('title',)
+            # note: no ordering_fields and serializer_class speficied
+
+            def get_serializer_class(self):
+                return OrderingFilterSerializer
+
+        view = OrderingListView.as_view()
+        request = factory.get('/', {'ordering': 'text'})
+        response = view(request)
+        self.assertEqual(
+            response.data,
+            [
+                {'id': 1, 'title': 'zyx', 'text': 'abc'},
+                {'id': 2, 'title': 'yxw', 'text': 'bcd'},
+                {'id': 3, 'title': 'xwv', 'text': 'cde'},
+            ]
+        )
+
+    def test_ordering_with_improper_configuration(self):
+        class OrderingListView(generics.ListAPIView):
+            queryset = OrderingFilterModel.objects.all()
+            filter_backends = (filters.OrderingFilter,)
+            ordering = ('title',)
+            # note: no ordering_fields and serializer_class
+            # or get_serializer_class speficied
+
+        view = OrderingListView.as_view()
+        request = factory.get('/', {'ordering': 'text'})
+        # BUG: I think this should raise ImproperlyConfigured
```

Comment:
@xordoquy Sorry, I realized that the `get_serializer_class` method actually has its own assert to check for `serializer_class`: https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/generics.py#L123.

On that note, is it better to get the error from `get_serializer_class` (`AssertionError`) or from `remove_invalid_fields` (`ImproperlyConfigured`)? I can rewrite the patch to either catch the `AssertionError` in `get_serializer_class` or let it pass right up.